### DISCUSSION TITLE
Roborock setup

### DISCUSSION THREAD

**Discussion by FilipeOS** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#696] [Q&A]
Hi guys,

So I'm new to hacs and I'm struggling with setup of my 2 Roborock S5 (I have 1 each floor).

I've setup the Xiaomi Miot Auto and the Vacuum Map Card in Frontend.

When I try to add the custom card vacuum to my dashboard I get the error map source
(Ps: what should I put in camera entity too?)

My mais question is how to make 2 vaccums work since have different tokens etc

Any advise please?

**Comment by sca075** [comment]
Well without a camera of Xaiomi Miot Auto should be listed between the camera in the developer tools  / states tab so far I understood from the integration setup you might have in customize.yaml something like:
```
camera.your_entity_id:
  video_attribute: 1   # https://github.com/al-one/hass-xiaomi-miot/issues/11#issuecomment-773054167
  keep_streaming: true # Continuously update stream address
```
the camera entity can be setup in the card very easily if correctly configured.

**Reply by FilipeOS** [reply]
Yes I noticed that I had to add in my configuration file the camera with the host and tokens correct? or not needed?

But I still think this is duplicated because I already have the integration and when I try to add both vacuums in configuration file it says that the host is not defined...

Since I posted this thread I noticed that I had to install the map extractor? Because I really don't know where to place your sample code because something seems to be missing. And the fact it's to vacuums it's confusing me more

**Comment by FilipeOS** [comment]
Saved by this video https://www.youtube.com/watch?v=dZmjyMfJnCU
and a thread to add  names to haves multiple cameras.. thank you.
